imap/http_prometheus.c

```diff
@@ -138,6 +126,9 @@ static int prom_get(struct transaction_t *txn,
     if (strcmp(txn->req_uri->path, "/metrics"))
         return HTTP_NOT_FOUND;
 
+    if (!httpd_userisadmin || !httpd_userid || httpd_userisanonymous)
+        return HTTP_UNAUTHORIZED;
+
```

Comment:
This check is breaking the `prometheus_need_auth` imapd.conf option when set to "user" or "none", which would be handled by the lines below, except that we already returned here if the user wasn't an admin.

> prometheus_need_auth: admin
> Authentication level required to fetch Prometheus metrics.
> 
> Allowed values: none, user, admin

I think you can just delete these new lines.  When I fetch your branch and do this locally, the Prometheus tests start passing again.